These patches build the label_image and benchmark_model tools with the
GPU delegate enabled via OpenGL.

diff --git a/tensorflow/lite/BUILD b/tensorflow/lite/BUILD
index 41bd12ab..decabb0f 100644
--- a/tensorflow/lite/BUILD
+++ b/tensorflow/lite/BUILD
@@ -268,6 +268,7 @@ cc_library(
         ":string",
         ":type_to_tflitetype",
         ":util",
+        "//tensorflow/lite/delegates/gpu:delegate",
         "//tensorflow/lite/delegates/nnapi:nnapi_delegate",
         "//tensorflow/lite/nnapi:nnapi_implementation",
         "//tensorflow/lite/c:common",
@@ -900,6 +901,8 @@ tflite_cc_shared_object(
         "//conditions:default": [
             "-Wl,-z,defs",
             "-Wl,--version-script,$(location //tensorflow/lite:tflite_version_script.lds)",
+            "-lEGL",
+            "-lGLESv2",
         ],
     }),
     per_os_targets = True,
diff --git a/tensorflow/lite/examples/label_image/BUILD b/tensorflow/lite/examples/label_image/BUILD
index 1c770a83..7cd7a13b 100644
--- a/tensorflow/lite/examples/label_image/BUILD
+++ b/tensorflow/lite/examples/label_image/BUILD
@@ -25,7 +25,10 @@ cc_binary(
             "-lm",  # some builtin ops, e.g., tanh, need -lm
             "-Wl,-rpath=/data/local/tmp",  # for hexagon delegate
         ],
-        "//conditions:default": [],
+        "//conditions:default": [
+            "-lEGL",
+            "-lGLESv2",
+        ],
     }),
     deps = [
         ":bitmap_helpers",
@@ -52,7 +55,9 @@ cc_binary(
             "//tensorflow/lite/delegates/gpu:delegate",
             "//tensorflow/lite/delegates/hexagon:hexagon_delegate",
         ],
-        "//conditions:default": [],
+        "//conditions:default": [
+            "//tensorflow/lite/delegates/gpu:delegate",
+        ],
     }),
 )
 
@@ -81,7 +86,9 @@ cc_library(
             "//tensorflow/lite/delegates/gpu:delegate",
             "//tensorflow/lite/delegates/hexagon:hexagon_delegate",
         ],
-        "//conditions:default": [],
+        "//conditions:default": [
+            "//tensorflow/lite/delegates/gpu:delegate",
+        ],
     }),
 )
 
diff --git a/tensorflow/lite/tools/benchmark/BUILD b/tensorflow/lite/tools/benchmark/BUILD
index 815efb77..5e1455b8 100644
--- a/tensorflow/lite/tools/benchmark/BUILD
+++ b/tensorflow/lite/tools/benchmark/BUILD
@@ -36,7 +36,10 @@ cc_binary(
             "-lm",  # some builtin ops, e.g., tanh, need -lm
             "-Wl,--rpath=/data/local/tmp/",  # Hexagon delegate libraries should be in /data/local/tmp
         ],
-        "//conditions:default": [],
+        "//conditions:default": [
+            "-lEGL",
+            "-lGLESv2",
+        ],
     }),
     tags = ["builder_default_android_arm64"],
     deps = [
@@ -191,7 +194,9 @@ cc_library(
         "//tensorflow:android": [
             "//tensorflow/lite/delegates/gpu:delegate",
         ],
-        "//conditions:default": [],
+        "//conditions:default": [
+            "//tensorflow/lite/delegates/gpu:delegate",
+        ],
     }),
 )
 
diff --git a/tensorflow/lite/tools/delegates/BUILD b/tensorflow/lite/tools/delegates/BUILD
index 89748679..e5ca13a9 100644
--- a/tensorflow/lite/tools/delegates/BUILD
+++ b/tensorflow/lite/tools/delegates/BUILD
@@ -66,12 +66,8 @@ cc_library(
     deps = [
         ":delegate_provider_hdr",
         "//tensorflow/lite/tools/evaluation:utils",
+        "//tensorflow/lite/delegates/gpu:delegate",
     ] + select({
-        "//tensorflow/lite/delegates/gpu:supports_gpu_delegate": [
-            "//tensorflow/lite/delegates/gpu:delegate",
-        ],
-        "//conditions:default": [],
-    }) + select({
         "//tensorflow:ios": [
             "//tensorflow/lite/delegates/gpu:metal_delegate",
         ],
@@ -103,6 +99,7 @@ cc_library(
     deps = [
         ":delegate_provider_hdr",
         "//tensorflow/lite/tools/evaluation:utils",
+        "//tensorflow/lite/delegates/gpu:delegate",
     ] + select({
         "//tensorflow:android_arm": [
             "//tensorflow/lite/delegates/hexagon:hexagon_delegate",
@@ -130,6 +127,7 @@ cc_library(
     deps = [
         ":delegate_provider_hdr",
         "//tensorflow/lite/tools/evaluation:utils",
+        "//tensorflow/lite/delegates/gpu:delegate",
     ] + select({
         "//tensorflow:ios": [
             "//tensorflow/lite/delegates/coreml:coreml_delegate",
@@ -151,6 +149,7 @@ cc_library(
     deps = [
         ":delegate_provider_hdr",
         "//tensorflow/lite/tools/evaluation:utils",
+        "//tensorflow/lite/delegates/gpu:delegate",
     ],
     alwayslink = 1,
 )
diff --git a/tensorflow/lite/tools/evaluation/BUILD b/tensorflow/lite/tools/evaluation/BUILD
index bb0d3ac4..13f6160b 100644
--- a/tensorflow/lite/tools/evaluation/BUILD
+++ b/tensorflow/lite/tools/evaluation/BUILD
@@ -42,12 +42,8 @@ cc_library(
     deps = [
         "//tensorflow/lite/c:common",
         "//tensorflow/lite/delegates/nnapi:nnapi_delegate",
+        "//tensorflow/lite/delegates/gpu:delegate",
     ] + select({
-        "//tensorflow/lite/delegates/gpu:supports_gpu_delegate": [
-            "//tensorflow/lite/delegates/gpu:delegate",
-        ],
-        "//conditions:default": [],
-    }) + select({
         "//tensorflow:android_arm": [
             "//tensorflow/lite/delegates/hexagon:hexagon_delegate",
         ],
